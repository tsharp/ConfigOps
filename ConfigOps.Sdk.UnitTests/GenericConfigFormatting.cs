using ConfigOps.Core.Serialization;
using ConfigOps.Sdk.Api.v1alpha1;
using Snapshooter.Xunit;

namespace ConfigOps.Sdk.UnitTests
{
    public class GenericConfigFormatting
    {
        [Fact]
        public void GenericConfig_Serialize()
        {
            var config = new GenericConfig
            {
                Metadata =
                {
                    Name = "global-config",
                    Namespace = "global"
                },
                Spec =
                {
                    InheritedConfigs = { 
                        "global/app-config-base-1" 
                    },
                    Settings =
                    {
                        { "setting1", "value1" },
                        { "setting2", "value2" }
                    }
                }
            };

            config
                 .SerializeToYaml()
                 .MatchSnapshot();
        }
    }
}
