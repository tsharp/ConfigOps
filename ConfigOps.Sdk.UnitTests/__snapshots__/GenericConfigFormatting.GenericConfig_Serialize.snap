apiVersion: config.ops.kuipersys.com/v1alpha1
kind: GenericConfig
metadata:
  name: global-config
  namespace: global
spec:
  inheritedConfigs:
  - global/app-config-base-1
  settings:
    setting1: value1
    setting2: value2
